src/components/Community/CommunitySidebar.tsx

```diff
@@ -0,0 +1,214 @@
+import { useState } from "react";
+
+import { styled } from "styled-components";
+
+import {
+  ImgHouse,
+  ImgWorkshop,
+  ImgProject,
+  ImgCommuity,
+  ImgContents,
+  ImgDesign,
+  ImgOpen,
+  IcArrowdownSm,
+} from "../../assets";
+
+import Dropdown from "./Dropdown";
+
+function CommunitySidebar() {
+  const navCategory = [
+    { content: "홈", element: <ImgHouse />, dropdown: false },
+    { content: "워크숍&세미나", element: <ImgWorkshop />, dropdown: false },
+    { content: "프로젝트&챌린지", element: <ImgProject />, dropdown: false },
+    { content: "커뮤니티", element: <ImgCommuity />, dropdown: true },
+    { content: "콘텐츠", element: <ImgContents />, dropdown: false },
+  ];
+
+  const outLinkCategory = [
+    { content: "디자이너 생존 테스트", element: <ImgDesign /> },
+    { content: "디자이너 오픈 채팅방", element: <ImgOpen /> },
+  ];
+
+  const commuityCategory = [
+    "전체",
+    "자유주제",
+    "포트폴리오",
+    "취업/이직/커리어",
+    "사이드 프로젝트/스터디",
+    "내 생존 유형은?",
+  ];
+
+  const [btnActive, setBtnActive] = useState(0);
+  const [toggleOpen, setToggleOpen] = useState(false);
+
+  const navBtnActive = (e: number) => {
+    setBtnActive(e);
+  };
+
+  const toggleBtnActive = () => {
+    setToggleOpen(!toggleOpen);
+  };
+
+  return (
+    <CommunutySidebarWrapper>
+      <SidebarContainer>
+        {navCategory.map((item, idx) => (
+          <SidebarBtnContainer key={item.content}>
+            <SidebarBtn
+              type="button"
+              key={item.content}
+              value={idx}
+              className={"btn" + (idx == btnActive ? " active" : "")}
+              // 현재는 홈에 바탕색 고정. 클릭마다 바탕색 이벤트를 주고 싶을 경우 0 >> idx로 변경
+              onClick={() => {
+                navBtnActive(0);
+                {
+                  item.dropdown === true ? toggleBtnActive() : null;
+                }
+              }}>
+              <NavImg>{item.element}</NavImg>
+              {item.content}
+              {item.dropdown === true ? (
+                <ToggleBtn type="button" className={"btn" + (toggleOpen == true ? " active" : "")} />
```

Comment:
Styled-Component는 prop을 넘겨서 style 처리를 할 수 있어요!! 이렇게 하는 방식보다 toggleOpen을 prop으로 넘겨주는 방식이 CSS-in-JS 방식에 더 맞는 것 같아서 이렇게 바꿔보는건 어떨까요~?
```suggestion
                <ToggleBtn type="button" $toggleOpen={toggleOpen} />
```

이렇게 prop으로 내려주면

```tsx
const ToggleBtn = styled(IcArrowdownSm)`
  position: absolute;
  right: 0;
  bottom: 1.9rem;
  transform: ${({ $toggleOpen }) => $toggleOpen && "rotate(180deg)"};
`;
```

이렇게 해줄 수 있어요!!

Reply:
저도 이 방법이 더욱 좋아보여요!